### PR TITLE
Update Workflows / Dependencies

### PR DESCRIPTION
**Pull request by syoon2**
This PR:

- Disables caching by `actions/setup-java`  
  `gradle/build-gradle-action` provides a finer-grain caching (enabled by default) than `actions/setup-java` does.
- Configures [Dependabot](https://docs.github.com/en/code-security/dependabot/dependabot-version-updates/about-dependabot-version-updates) for automated dependency updates
- Manually bumps up dependencies
  - Bumps up `commons-lang3` from 3.12.0 to 3.13.0
  - Bumps up JUnit 5 dependencies from 5.9.3 to 5.10.0
  - Bumps up `guava` from 23.0 to 32.1.2-jre
  - Bumps up `spotless` from 6.18.0 to 6.20.0 (and reapplies coding style with the updated version)

### CODE REVIEW

File: .github/dependabot.yml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+version: 2
+updates:
+  - package-ecosystem: "github-actions" # See documentation for possible values
+    directory: "/" # Location of package manifests
+    schedule:
+      interval: "daily"
```
Comment (williamfiset) [inline .github/dependabot.yml:6]: Would you mind updating these intervals to be less frequent to reduce PR noise in the event of frequent dependency updates. Perhaps weekly or every two weeks?

**Comment by williamfiset** [comment]
I had no idea tools like Dependabot existed, thanks for the PR! Reading through the docs it looks like it the tool itself will send a pr to update the deps to newer versions, that's super cool! Just added one minor comment about the frequency of the updates

File: .github/dependabot.yml (hunk 1 above)
Comment (syoon2) [reply .github/dependabot.yml]: It's now updated to run on a weekly schedule. 
See e0f1b7b :-)

**Comment by syoon2** [comment]
> Would you mind updating these intervals to be less frequent to reduce PR noise in the event of frequent dependency updates. Perhaps weekly or every two weeks?

It's now updated to run on a weekly schedule :smile: - see e0f1b7b

**Pull request merged**